File: Arrays & Hashing/8. Longest Consecutive Sequence/solution.rb
# @param {Integer[]} nums
# @return {Integer}
def longest_consecutive(nums)
  numSet = Set.new(nums)
  longest = 0

  for n in nums
    if !numSet.include?(n - 1)
      length = 0
      while numSet.include?(n + length)
        length += 1
      end
      longest = [length, longest].max
    end
  end

  longest
end

File: Arrays & Hashing/8. Longest Consecutive Sequence/solution.scala
object Solution {
    def longestConsecutive(nums: Array[Int]): Int = {
       val numsSet = Set.from(nums)
       var longest = 0

        for(n <- nums) {
            if(!numsSet.contains(n - 1)) {
                var length = 0
                while(numsSet.contains(n + length)) {
                    length += 1
                }
                longest = longest.max(length)
            }
        }

        longest
    }
}
